src/pretix/presale/views/organizer.py

```diff
@@ -719,6 +735,331 @@ def _events_by_day(self, before, after):
+        tick_duration = next((d for d in [5, 10, 15, 30, 60, 120, 180] if d >= shortest_duration), 180)
+
+        raster_size = min(self._get_raster_size(events), tick_duration)
+        events, start, end = self._rasterize_events(events, tick_duration=tick_duration, raster_size=raster_size)
+        calendar_duration = self._get_time_duration(start, end)
+        ctx["calendar_duration"] = self._format_duration(calendar_duration)
+        ctx['time_ticks'] = self._get_time_ticks(start, end, tick_duration)
+        ctx['start'] = datetime.combine(self.date, start)
+        ctx['raster_size'] = raster_size
+        # ctx['end'] = end
+        # size of each grid-column in calendar is based on shortest event duration
+        ctx['shortest_duration_rastered'] = next((d for d in [5, 10, 15, 30, 60, 90, 120, 180] if d >= shortest_duration), 180)
+
+        ctx['events'] = events
+
+        events_by_series = self._grid_for_template(events)
+        ctx['collections'] = events_by_series
+        ctx['no_headlines'] = not any([series for series, events in events_by_series])
+        ctx['multiple_timezones'] = self._multiple_timezones
+        return ctx
+
+    # get best raster-size for min. # of columns in grid
+    # due to grid-col-calculations in CSS raster_size cannot be bigger than 60 (minutes)
+    def _get_raster_size(self, events):
+        # all start- and end-times (minute-part) except full hour
+        times = [
+            e["time"].minute for e in events if e["time"] and e["time"].minute
+        ] + [
+            e["time_end_today"].minute for e in events if "time_end_today" in e and e["time_end_today"] and e["time_end_today"].minute
+        ]
+        if not times:
+            # no time other than full hour, so raster can be 1 hour/60 minutes
+            return 60
+        gcd = reduce(math.gcd, set(times))
+        return next((d for d in [5, 10, 15, 30, 60] if d >= gcd), 60)
+
+    def _get_time_duration(self, start, end):
+        midnight = time(0, 0)
+        return datetime.combine(
+            self.date if end != midnight else self.date + timedelta(days=1),
+            end
+        ) - datetime.combine(
+            self.date,
+            start
+        )
+
+    def _format_duration(self, duration):
+        return ":".join([
+            "%02d" % i for i in (
+                (duration.days * 24) + (duration.seconds // 3600),
+                (duration.seconds // 60) % 60
+            )
+        ])
+
+    # currently based on minutes, might be factored into a helper class with a timedelta as raster?
+    def _floor_time(self, t, raster_size=5):
+        minutes = t.hour * 60 + t.minute
+        if minutes % raster_size:
+            minutes = (minutes // raster_size) * raster_size
+            return t.replace(hour=minutes // 60, minute=minutes % 60)
+        return t
+
+    # currently based on minutes, might be factored into a helper class with a timedelta as raster?
```

Comment:
I think it's generally more common in Python and less "risky" to put comments about the function **inside** the function, not before them, even if they're not actual "docstrings".

Reply:
Thanks for the hint, I will put them inside.